src/main/java/com/forgeessentials/api/UserIdent.java

```diff
@@ -129,11 +131,89 @@ private UserIdent(UUID identUuid, String identUsername, EntityPlayerMP identPlay
                 byUuid.put(this.uuid, this);
             if (identUsername != null && identUsername.charAt(0) != '@')
                 byUsername.put(identUsername.toLowerCase(), this);
-            
-            // TODO: Contact Mojang servers to get UUID if null?
+
+            if (uuid == null && username != null)
+                uuid = resolveMissingUUID(username);
+            else if (uuid != null && username == null)
+                username = resolveMissingUsername(uuid);
         }
     }
 
+    public static UUID hexStringToUUID(String s)
```

Comment:
Think you can move all that Mojang related stuff into a utility class or at least a separate section (marked by large line comments) at the end of the file?
It would be nice to keep the top of UserIdent for it's actual main features to have some structure in it.